indico/util/notification_templates.py

```diff
@@ -0,0 +1,113 @@
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from sqlalchemy.dialects.postgresql import ARRAY, JSON
+from sqlalchemy.ext.declarative import declared_attr
+
+from indico.core.db import db
+
+
+class NotificationTemplateBase(object):
+    """Base class for notification templates."""
+
+    #: The ID of the template
+    @declared_attr
+    def id(cls):
+        return db.Column(
+            db.Integer,
+            primary_key=True
+        )
+
+    #: The ID of the event
+    @declared_attr
+    def event_id(cls):
+        return db.Column(
+            db.Integer,
+            db.ForeignKey('events.events.id'),
+            index=True,
+            nullable=False
+        )
+
+    #: The title of the template
+    @declared_attr
+    def title(cls):
+        return db.Column(
+            db.String,
+            nullable=False
+        )
+
+    #: The address to use as Reply-To in the notification email
+    @declared_attr
+    def reply_to_address(cls):
+        return db.Column(
+            db.String,
+            nullable=False
+        )
+
+    #: The subject of the notification email
+    @declared_attr
+    def subject(cls):
+        return db.Column(
+            db.String,
+            nullable=False
+        )
+
+    #: The body of the notification template
+    @declared_attr
+    def body(cls):
+        return db.Column(
+            db.Text,
+            nullable=False,
+            default=''
+        )
+
+    #: Whether to stop checking the rest of the conditions when a match is found
+    @declared_attr
+    def stop_on_match(cls):
+        return db.Column(
+            db.Boolean,
+            nullable=False,
+            default=True
+        )
+
+    #: Conditions need to be met to send the notification email
+    @declared_attr
+    def conditions(cls):
+        return db.Column(
+            JSON,
+            nullable=False
+        )
+
+    #: List of extra email addresses to be added as CC in the notification email
+    @declared_attr
+    def extra_cc_emails(cls):
+        return db.Column(
+            ARRAY(db.String),
+            nullable=True,
+            default=[],
+        )
+
+    #: List of email addresses to send the notification email
+    @property
+    def to_emails(self):
+        raise NotImplementedError
+
+    #: List of email addresses to be added as CC in the notification email
```

Comment:
So, `"""..."""` instead of `:# ...`?
Then we should change here at some point too :stuck_out_tongue: 
https://github.com/indico/indico/blob/master/indico/core/storage/models.py#L62